apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1/MetadataClientImpl.cs

```diff
@@ -0,0 +1,702 @@
+        /// </para>
+        /// <para>
+        /// The key must conform to the following regular expression: [a-zA-Z0-9-_]+ and must be less than 128 bytes in length.
+        /// </para>
+        /// </remarks>
+        /// <param name="key">The key of the project's custom metadata value to get.</param>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        /// <exception cref="ArgumentException">The key is not in the proper format.</exception>
+        /// <returns>A task representing the asynchronous operation.</returns>
+        /// <seealso cref="Project.Metadata"/>
+        public override async Task<string> GetCustomProjectMetadataAsync(string key, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            ValidateCustomMetadataKey(key);
+            var response = await RequestMetadata("project/attributes/" + key, cancellationToken).ConfigureAwait(false);
+            return await response.Content.ReadAsStringAsync().ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Gets the metadata for the VM currently running this code synchronously.
+        /// </summary>
+        /// <remarks>
+        /// The instance metadata contains information about the instance, such as the host name, instance ID, and custom metadata and it can be
+        /// accessed without authentication from an application running on the instance. Check <see cref="IsServerAvailable"/> or
+        /// <see cref="IsServerAvailableAsync(CancellationToken)"/> to determine if the metadata can be accessed.
+        /// </remarks>
+        /// <returns>The <see cref="Instance"/> representing the metadata.</returns>
+        public override Instance GetInstanceMetadata() =>
+            GetResult(() => GetInstanceMetadataAsync());
+
+        /// <summary>
+        /// Gets the metadata for the VM currently running this code asynchronously.
+        /// </summary>
+        /// <remarks>
+        /// The instance metadata contains information about the instance, such as the host name, instance ID, and custom metadata and it can be
+        /// accessed without authentication from an application running on the instance. Check <see cref="IsServerAvailable"/> or
+        /// <see cref="IsServerAvailableAsync(CancellationToken)"/> to determine if the metadata can be accessed.
+        /// </remarks>
+        /// <param name="cancellationToken">The token to monitor for cancellation requests.</param>
+        /// <returns>A task containing the <see cref="Instance"/> representing the metadata.</returns>
+        public override async Task<Instance> GetInstanceMetadataAsync(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            var response = await RequestMetadata("instance?recursive=true", cancellationToken).ConfigureAwait(false);
+            dynamic obj = JObject.Parse(await response.Content.ReadAsStringAsync().ConfigureAwait(false));
```

Comment:
Is the JSON sufficiently different from this that we can't just get JsonConvert to deserialize it? (That's fine if that's the case...)